compiler/ballerina-lang/src/main/resources/compiler.properties

```diff
@@ -476,6 +476,9 @@ error.invalid.xml.ns.interpolation=\
 error.iterable.not.supported.collection=\
   incompatible types: ''{0}'' is not an iterable collection
 
+error.iterable.not.supported.object=\
+  incompatible types: ''{0}'' is not an iterable object
```

Comment:
We can improve the error message saying why it is not iterable. There are two possibilities. 
1) __iterator function not found. 
2) __iterator function signature not matched.

Reply:
fixed